src/eventmaster/plugins.go

```diff
@@ -0,0 +1,84 @@
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/sha1"
+	"crypto/subtle"
+	"encoding/hex"
+	"encoding/json"
+	"net/http"
+	"time"
+)
+
+var Plugins = map[string]Plugin{}
+
+type Plugin interface {
+	ParseRequest(*http.Request, []byte) ([]*UnaddedEvent, error)
+	ValidateAuth(*http.Request, []byte) bool
+}
+
+type GitHubPlugin struct {
```

Comment:
Should have a mechanism to validate config before getting an event (which would allow it to not start if the config was bad) -- this can be done as a custom json unmarshal, or as another step in the plugin interface (which would let it open connections etc. if it did that sort of thing)